In [ ]:
import numpy as np
from scipy import stats

###Q1:- Given a portfolio’s VaR at 95% confidence is 10,000, compute the Expected Shortfall (ES) if the tail loss distribution follows a normal distribution.

In [ ]:
VaR_95 = 10000  # Given 95% VaR
z_95 = stats.norm.ppf(0.05)  # Z-score for 5% left tail (95% confidence)
phi_z = stats.norm.pdf(z_95)
alpha = 0.05
ES_95 = (VaR_95 * phi_z) / alpha
print(f"\nExpected Shortfall (95%) assuming normal distribution: ${ES_95:,.2f}")


Expected Shortfall (95%) assuming normal distribution: $20,627.13


If the things got really bad then max 20,657.13 is the max amount we lose.

###Q2:- A portfolio has an expected annual return of 12%, a risk-free rate of 2%, and a standard deviation of 18%. Compute the Sharpe Ratio.

In [ ]:
expected_return = 0.12
risk_free_rate = 0.02
sigma = 0.18

sharpe_ratio = (expected_return - risk_free_rate) / sigma
print(f"Sharpe Ratio: {sharpe_ratio:.3f}")

Sharpe Ratio: 0.556


It tells us that 0.556 is a return to risk taken ratio

###Q3:- Given mean return = 7%, volatility = 12%, and time horizon = 1 year, simulate one possible future portfolio return using a Monte Carlo model.

In [ ]:
mu = 0.07
volatility = 0.12
Z = np.random.normal()
print(Z)
simulated_return = mu + volatility * Z
print(f"Simulated 1-Year Return (Monte Carlo): {simulated_return:.2%}")

0.1678428158487803
Simulated 1-Year Return (Monte Carlo): 9.01%


It show us the future capabilities of portfolio